frontend/public/components/utils/dropdown.jsx

```diff
@@ -285,7 +285,18 @@ class Dropdown_ extends DropdownMixin {
   applyTextFilter_(autocompleteText, items) {
     const { autocompleteFilter } = this.props;
     if (autocompleteFilter && !_.isEmpty(autocompleteText)) {
-      items = _.pickBy(items, (item, key) => autocompleteFilter(autocompleteText, item, key));
+      const newItems = [];
+      let totalItems = _.map(items, (item) => item.props);
+      totalItems = autocompleteFilter(autocompleteText, totalItems);
+      _.map(totalItems, (filterItem) => {
+        _.some(items, (item) => {
```

Comment:
I would avoid nested loops here for performance. Sometimes these lists are large, and the sort is reevaluated every keystroke. Doesn't fuzzysearch return the original object? Do we need to find it again in the items list? If there is no other way, it would be better to create a map keyed by name so we don't have to iterate over the list each time.